### PR TITLE
test: refactor integration tests to unit tests

### LINKED ISSUE #596: migrate existing integration tests as unit tests where possible (increase unit test coverage)

### PR DESCRIPTION
**Pull request by jyeshe**
Closes #596

### CODE REVIEW

File: test/ae_mdw/db/sync/contract_test.exs [hunk 1]
```diff
@@ -92,5 +95,49 @@ defmodule AeMdw.Db.Sync.ContractTest do
...
+      call_txi = 23_198_023
+
+      event_mutations =
+        "AENS.update"
+        |> contract_events()
+        |> Contract.events_mutations(block_index, <<>>, call_txi, <<>>, -1)
+        |> List.flatten()
+
+      assert Enum.any?(event_mutations, fn
+               %NameUpdateMutation{
```
Comment (jyeshe) [inline test/ae_mdw/db/sync/contract_test.exs:132]: same

File: test/ae_mdw/db/sync/transaction_test.exs [hunk 2]
```diff
@@ -84,9 +87,107 @@ defmodule AeMdw.Db.Sync.TransactionTest do
     end
   end
 
+  describe "transaction_mutations/4" do
+    test "aex9 contract creation by a contract call" do
+      signed_tx =
+        {:signed_tx,
```
Comment (jyeshe) [inline test/ae_mdw/db/sync/transaction_test.exs:93]: real chain signed_tx

File: test/ae_mdw/db/sync/transaction_test.exs [hunk 3]
```diff
@@ -84,9 +87,107 @@ defmodule AeMdw.Db.Sync.TransactionTest do
...
+
+      with_mocks [
+        {Contract, [],
+         [
+           is_aex9?: fn ct_pk -> ct_pk != contract_pk end,
+           aex9_meta_info: fn ct_pk -> if ct_pk != contract_pk, do: {:ok, aex9_meta_info} end,
+           call_tx_info: fn _tx, ^contract_pk, _block_hash, _to_map ->
+             {
+               fun_args_res("create_pair"),
+               call_rec("create_pair")
```
Comment (jyeshe) [inline test/ae_mdw/db/sync/transaction_test.exs:140]: chain data

File: test/ae_mdw/db/sync/contract_test.exs [hunk 4]
```diff
@@ -92,5 +95,49 @@ defmodule AeMdw.Db.Sync.ContractTest do
...
+      call_txi = 11_684_918
+
+      event_mutations =
+        "AENS.transfer"
+        |> contract_events()
+        |> Contract.events_mutations(block_index, <<>>, call_txi, <<>>, -1)
+        |> List.flatten()
+
+      assert Enum.any?(event_mutations, fn
+               %NameTransferMutation{
```
[the remark this answers is not in the document]
Comment (jyeshe) [reply test/ae_mdw/db/sync/contract_test.exs:163]: `NameTransferMutation` shall have it's own test case (not part of the scope of this PR)

File: test/ae_mdw/db/sync/contract_test.exs (hunk 1 above)
Comment (jyeshe) [reply test/ae_mdw/db/sync/contract_test.exs:185]: same

**Review by thepiwo — APPROVED** [review]

**Pull request merged**